src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -112,21 +124,41 @@ private StorageListItem GetStorageListItem(IListBlobItem listBlobItem)
                     }
 
                     destinationStream.Seek(0, SeekOrigin.Begin);
-                    await blob.UploadFromStreamAsync(destinationStream, cancellationToken);
+                    await blob.UploadFromStreamAsync(destinationStream,
+                                                     accessCondition: null,
+                                                     options: _blobRequestOptions,
+                                                     operationContext: null,
+                                                     cancellationToken: cancellationToken);
                     Trace.WriteLine(String.Format("Saved compressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
                 }
             }
             else
             {
                 using (Stream stream = content.GetContentStream())
                 {
-                    await blob.UploadFromStreamAsync(stream, cancellationToken);
+                    await blob.UploadFromStreamAsync(stream,
+                                                     accessCondition: null,
+                                                     options: _blobRequestOptions,
+                                                     operationContext: null,
+                                                     cancellationToken: cancellationToken);
                     Trace.WriteLine(String.Format("Saved uncompressed blob {0} to container {1}", blob.Uri.ToString(), _directory.Container.Name));
                 }
             }
             await TryTakeBlobSnapshotAsync(blob);
         }
 
+        private BlobRequestOptions CreateBlobRequestOptions(int maxExecutionTimeInSeconds)
+        {
+            int internalMaxTimeout = maxExecutionTimeInSeconds <= 0 ? _defaultMaxTimeoutInSeconds : maxExecutionTimeInSeconds;
+           
+            return new BlobRequestOptions
+            {
+                ServerTimeout = TimeSpan.FromSeconds(_serverTimeoutSeconds),
+                MaximumExecutionTime = TimeSpan.FromSeconds(internalMaxTimeout),
+                RetryPolicy = new ExponentialRetry()
```

Comment:
Does this just exponentially back off until `MaximumExecutionTime` is exhausted?